core/src/main/java/com/sequenceiq/cloudbreak/aspect/SecretAspects.java

```diff
@@ -162,12 +162,14 @@ private Object proceedDelete(ProceedingJoinPoint proceedingJoinPoint) {
     }
 
     private String findTenant(Object entity) {
-        return Optional.ofNullable(entity)
-                .filter(e -> e instanceof TenantAwareResource)
-                .map(e -> (TenantAwareResource) e)
-                .map(TenantAwareResource::getTenant)
-                .map(Tenant::getName)
-                .orElseThrow(() -> new IllegalArgumentException(
-                        entity.getClass().getSimpleName() + " must be a subclass of " + TenantAwareResource.class.getSimpleName()));
+        if (entity instanceof TenantAwareResource) {
+            return ((TenantAwareResource) entity).getTenant().getName();
+        } else if (entity instanceof AccountIdAwareResource) {
+            return ((AccountIdAwareResource) entity).getAccountId();
```

Comment:
these don't handle if entity is null, like the old logic did (`Optional.ofNullable`)

Reply:
Acknowledged, will update the logic